bloom-housing/bloom
# feat: feature flag consumption

This PR addresses [#(4459)](https://github.com/bloom-housing/bloom/issues/4459)


## Description

Creates the module, controller and service files for the new Feature Flag table. Allows for standard CRUD methods, as well as a list endpoint and an endpoint to add/remove jurisdiction associations with the flag (`associateJurisdictions`). Permissions are limited to admin only. Flags associated with jurisdictions will appear when jurisdiction data is returned.

## How Can This Be Tested/Reviewed?

Start up the backend. Login as an admin, either through the API or partners portal. Test each endpoint:
- Create a feature flag
- Get the feature flag using the id returned ^^
- Update the feature flag using the id
- Get the feature flag to verify update
- Associate jurisdiction(s) with the flag
- Get the feature flag to verify association
- Get the jurisdiction to verify association
- Call the list endpoint
- Remove the jurisdiction(s) using associateJurisdictions
- Get the feature flag to verify removal
- Get the jurisdiction to verify removal
- Delete feature flag
- Call list endpoint to verify

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
